### ISSUE TITLE
Docs only showing tags in details section of source objects

### ISSUE THREAD

**Issue by crystalro0** [dbt-labs/dbt-docs#488]
### Describe the bug
User is not seeing their sources details from their yaml file displayed in docs after doing a dbt docs generate on `1.6` and `1.7`.

### Steps To Reproduce
1. dev (or prod) env on `v1.7` or `1.6`
2. do `dbt docs generate`
3. Switch env to `1.5` then generate again, and the details section is populated.

### Expected behavior
<img width="757" alt="v1.5" src="https://github.com/dbt-labs/dbt-docs/assets/87022257/71cc0217-7c37-49ae-b8f9-c7a414d6de2e">

### Screenshots and log output
<img width="757" alt="v1.7" src="https://github.com/dbt-labs/dbt-docs/assets/87022257/ae99acc1-2d99-41cf-8f09-7a6ccaa7178f">

**The output of `dbt --version`:**
`1.6` and `1.7`

### Additional context
Displayed in dbt Explorer in `1.7`.

**Comment by dbeatty10** [comment]
Thanks for reporting this @crystalro0 !

I see what you are saying about 1.5 working but 1.6 and 1.7 having a bug.

### Reprex

Here's the `models/_sources.yml` that I used:

```yaml
version: 2

sources:
  - name: agdirect_dbo
    tables:
      - name: dim_customers
```

Then I ran these commands using dbt-core:
```
dbt docs generate
dbt docs serve
```

When I open the developer tools in Chrome, I can see the following javascript errors:

```
(index):73 TypeError: Cannot read properties of undefined (reading 'contract')
(index):73 TypeError: Cannot read properties of undefined (reading 'enforced')
```

<img width="1437" alt="image" src="https://github.com/dbt-labs/dbt-docs/assets/44704949/5c56e0ba-9f8e-4bfe-9b2d-6899cb45119c">

**Comment by crystalro0** [comment]
@dbeatty10 you're awesome! fixed my sandbox with:
```
version: 2

sources:

  - name: agdirect_dbo
    config:
      contract:
        enforced: false
    database: DEMO_DB
    loader: qlik_replicate
    tables:
      - name: dim_customers
```
Thank you!

> Thanks for reporting this @crystalro0 !
> 
> I see what you are saying about 1.5 working but 1.6 and 1.7 having a bug.
> 
> ### Reprex
> Here's the `models/_sources.yml` that I used:
> 
> ```yaml
> version: 2
> 
> sources:
>   - name: agdirect_dbo
>     tables:
>       - name: dim_customers
> ```
> 
> Then I ran these commands using dbt-core:
> 
> ```
> dbt docs generate
> dbt docs serve
> ```
> 
> When I open the developer tools in Chrome, I can see the following javascript errors:
> 
> ```
> (index):73 TypeError: Cannot read properties of undefined (reading 'contract')
> (index):73 TypeError: Cannot read properties of undefined (reading 'enforced')
> ```
> 
> <img alt="image" width="1437" src="https://private-user-images.githubusercontent.com/44704949/300581511-5c56e0ba-9f8e-4bfe-9b2d-6899cb45119c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY1NzU4NDksIm5iZiI6MTcwNjU3NTU0OSwicGF0aCI6Ii80NDcwNDk0OS8zMDA1ODE1MTEtNWM1NmUwYmEtOWY4ZS00YmZlLTliMmQtNjg5OWNiNDUxMTljLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTMwVDAwNDU0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWUyZjg0NTZjN2U4M2QwM2Y3YjFlN2QxNGQ5YzI5M2U5YTIyZGE1N2QzNGY0MmYyMDQ3NzA3ZWEwZDNmYzEwZWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.NFO7Ul0Y0XmQZumtYJ9Lqq4L7-mvZtVxIGCxdVW3K3E">

**Comment by dbeatty10** [comment]
Nice idea @crystalro0 -- adding the `contract` config is a clever workaround! 🧠

**Comment by dbeatty10** [comment]
This looks like it is fixed in v1.8.

I'm not sure if it was fixed in a [commit in dbt-docs](https://github.com/dbt-labs/dbt-docs/commits/main/) or [something within `dbt-core`](https://github.com/dbt-labs/dbt-core/blob/1.8.latest/CHANGELOG.md).

Either way, I'm closing this as resolved.
